src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestV4ServiceImpl.java

```diff
@@ -0,0 +1,401 @@
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4ReportDTO;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4StatusDTO;
+import edu.cornell.kfs.concur.service.ConcurAccountValidationService;
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+public class ConcurRequestV4ServiceImpl implements ConcurRequestV4Service {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    protected static final String PROCESSING_ERROR_MESSAGE = "Encountered an error while processing travel request";
+
+    protected ConcurBatchUtilityService concurBatchUtilityService;
+    protected ConcurEventNotificationV2WebserviceService concurEventNotificationV2WebserviceService;
+    protected ConcurAccountValidationService concurAccountValidationService;
+    protected ConfigurationService configurationService;
+    protected DateTimeService dateTimeService;
+
+    @Override
+    public List<ConcurEventNotificationProcessingResultsDTO> processTravelRequests(String accessToken) {
+        LOG.info("processTravelRequests, Starting processing of travel requests");
+        RequestV4QuerySettings querySettings = new RequestV4QuerySettings(
+                accessToken, dateTimeService.getCurrentDate(), buildUserIdFilter());
+        
+        List<ConcurEventNotificationProcessingResultsDTO> results = findAndProcessPendingTravelRequests(
+                querySettings, this::processTravelRequestsSubset);
+        
+        updateLastProcessedDateIfNecessary(querySettings);
+        
+        LOG.info("processTravelRequests, Finished processing " + results.size() + " travel requests");
+        return results;
+    }
+
+    protected Predicate<String> buildUserIdFilter() {
+        Map<String, String> testRequestUsers = getRequestV4TestUserMappings();
+        if (isProduction()) {
+            LOG.info("buildUserIdFilter, Only Production travel requests will be processed for this run");
+            return userId -> !testRequestUsers.containsKey(userId);
+        } else {
+            LOG.info("buildUserIdFilter, Only non-Production travel requests will be processed for this run");
+            return userId -> testRequestUsers.containsKey(userId);
+        }
+    }
+
+    protected <T> List<T> findAndProcessPendingTravelRequests(RequestV4QuerySettings querySettings,
+            BiFunction<RequestV4QuerySettings, ConcurRequestV4ListingDTO, Stream<T>> requestListingProcessor) {
+        Stream.Builder<Stream<T>> subResults = Stream.builder();
+        Stream<T> subResult;
+        String initialQueryUrl = buildInitialRequestQueryUrl(querySettings);
+        String currentQueryUrl = initialQueryUrl;
+        int page = 1;
+        ConcurRequestV4ListingDTO requestListing;
+        
+        do {
+            requestListing = findPendingTravelRequests(querySettings.accessToken, page, currentQueryUrl);
+            subResult = requestListingProcessor.apply(querySettings, requestListing);
```

Comment:
No, the "requestListingProcessor" is what takes the intermediate Request Listings, and iterates over their Requests to validate their accounting info and return the results. I had left this as a BiFunction since it simplified my setup of the unit tests that simply retrieve and validate the Request Listing objects. I agree that its setup is still a little confusing, though, so I'll see if I can simplify this part further and adjust the unit tests accordingly.